server/controllers/stock/reports/index.js

```diff
@@ -592,9 +592,29 @@ function stockLotsReport(req, res, next) {
   let options = {};
   let display = {};
   let hasFilter = false;
+  let filters = [];
 
   const data = {};
   let report;
+
+// getting params
+  var defaultParam = req.query;
+  try {
+
+    const keys = Object.keys(defaultParam);
+    keys.forEach((key) => {
+      if ((key !== 'lang') && (key !== 'renderer')) {
```

Comment:
Hmm.... this isn't really extensible.  Also, this method doesn't allow keys to be translated.

Why don't you try using something like like the `formatFilters()` function from the patient registration report?  It lists every key and it's translation value ... like this:
```js
// translation key mappings for dynamic filters
// Basically, to show a pretty filter bar, this will translate URL query params
// into human-readable text to be placed in the report, showing the properties
// filtered on.
function formatFilters(qs) {
  const columns = [
    { field : 'name', displayName : 'FORM.LABELS.NAME' },
    { field : 'sex', displayName : 'FORM.LABELS.GENDER' },
    { field : 'hospital_no', displayName : 'FORM.LABELS.HOSPITAL_NO' },
    { field : 'reference', displayName : 'FORM.LABELS.REFERENCE' },
    { field : 'dateBirthFrom', displayName : 'FORM.LABELS.DOB', comparitor : '>', isDate : true },
    { field : 'dateBirthTo', displayName : 'FORM.LABELS.DOB', comparitor : '<', isDate : true },
    { field : 'dateRegistrationFrom', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '>', isDate : true },
    { field : 'dateRegistrationTo', displayName : 'FORM.LABELS.DATE_REGISTRATION', comparitor : '<', isDate : true },
    { field : 'debtor_group_uuid', displayName : 'FORM.LABELS.DEBTOR_GROUP' },
    { field : 'patient_group_uuid', displayName : 'PATIENT_GROUP.PATIENT_GROUP' },
    { field : 'user_id', displayName : 'FORM.LABELS.USER' },
    { field : 'limit', displayName : 'FORM.LABELS.LIMIT' },
    { field : 'period', displayName : 'TABLE.COLUMNS.PERIOD', isPeriod : true },
  ];

  return columns.filter(column => {

    const value = qs[column.field];

    if (!_.isUndefined(value)) {
  
      if (column.isPeriod) {
        const service = new PeriodService(new Date());
        column.value = service.periods[value].translateKey;
      } else {
        column.value = value;
      }
      return true;
    }
    return false;
  });
} 
```

This format has the following advantages:
 1. It's clear which filters are supported and which are not.
 2. The display name can be anything found in the translation dictionary.
 3. No need to `try/catch` since the code will ignore anything that isn't in the supported filters object.  
 4. Returns an array as required.
 5. Allows any additional properties to be added to a given filter (like the `isDate` property or the `comparitor` property).